codes/classical/q-ary_digits/hermitian_codes.yml

```diff
@@ -0,0 +1,62 @@
+  curve. In fact, in affine coordinates the Stichtenoth form of the curve is \(f(x,y) = x^{q+1} - y^{q} - y = N(x) - Tr(y)\) where \(N\) is the 
+  \href{https://en.wikipedia.org/wiki/Field_norm}{field norm} and \(Tr\) is the \href{https://en.wikipedia.org/wiki/Field_trace}{Field Trace}. If viewed in the Fermat
+  version again the curve can be written as \(H(u,v,w) = u\overline{u} + v\overline{v} - w\overline{w}\) where the conjugation map 
+  \(\overline{u} = u^{q}\) is an isomorphism of \(\mathbb{F}_q \). In fact, when the field of evaluations \(\mathbf{F}_{q^2}\) is viewed as a quadratic extension of 
+  \(\mathbf{F}_q\) then the conjugation map is an \(\mathbf{F}_q\)-isomorphism that permutes the roots of the quadratic irreducible polynomial
+  used to generate \(\mathbf{F}_{q^2}\) from \( \mathbf{F}_q[x]\).   
+protection: 
+  - 'When \(D < q + 1 \) the Hermitian code protects against at least \(n - (q+1)J\) errors'
+  - 'When \(D \geq q+1 \) the Hermitian code protects against at least \(n-k - \frac{q(q-1)}{2} + 1\) errors'
+  
+features:
+
+  rate: 
+  - 'When \(D < q + 1 \), \(k = \frac{(D+1)(D+2)}{2}\)' 
+  - 'When \(D \geq q + 1 \), \(k = (q+1)D - \frac{q(q-1)}{2} + 1 \)' 
+  
+  encoders:
+    - 'Multivariate Polynomial Evaluation' 
+
+  decoders:
+    - 'Unique decoding using syndromes and error locator ideals: Note that Hermitian codes are linear codes so we can compute the syndrome of a received vector. Moreover, akin to the error-locator ideals found in decoding Reed-Solomon codes, for the multivariate case we must define an error locator ideal \(\Lambda \) such that the variety of this ideal over \(F^{2}_{q}\) is exactly the set of errors. The Sakata algorithm uses these two ingredients to get a unique decoding procedure \cite{manual: {Sakata, Shojiro. "Finding a minimal set of linear recurring relations capable of generating a given finite two-dimensional array." Journal of Symbolic Computation 5.3 (1988): 321-337}}'
+    - 'List decoding using Sudan's algorithm: \cite{manual: {Venkatesan Guruswami, Madhu Sudan. "Improved decoding of Reed-Solomon and algebraic-geometry codes." (1999).}}'
+
+notes: 
+  - 'Hermitian codes, directly improve Reed-Solomon codes in the sense that RS codes have length at most \(q\) while Hermitian codes have length \(q^3 + 1\)
+  - 'Hermitian codes are special examples of Algebraic geometry codes because Hermitian curves have the largest number of points that a curve of a fixed genus can have over a given finite field. So for a curve of fixed genus, Hermitian codes are the algebraic geometry codes which have the maximum possible length'
+  - 'Hermitian codes can also come in affine and epicyclic flavours, depending on whether the evaluations are over the affine plane or the bicyclic plane. The affine codes have length \(q^3 - q\) while epicyclic codes have length \((q-2)(q+1)^2\) 
+
+relations:
+  parents:
+    - code_id: binary_linear
```

Comment:
Hermitian codes are not binary, so "binary linear" are not their parents. Maybe q-ary linear.

Reply:
parent: evaluation